go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus-community/postgres_exporter
 
-go 1.14
+go 1.16
```

Comment:
In tools we have 1.17, can we use same version everywhere?

Reply:
It is min version, we can build this app with 1.18
We need to rise min version, when we have reason for that, right?

if you are referring to some best practices, can you please share link

Reply:
But it is a min version, you can build with 1.17, 1.18. 

Here is an example of running checks with 1.16.0:

```
➜  postgres_exporter git:(PMM-7806-upgrade-pg-exporter) ✗ go version
go version go1.16 linux/amd64
➜  postgres_exporter git:(PMM-7806-upgrade-pg-exporter) ✗ make common-all
>> checking code style
mkdir -p /home/am/go/1.16.0/bin
curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.42.0/install.sh \
        | sed -e '/install -d/d' \
        | sh -s -- -b /home/am/go/1.16.0/bin v1.42.0
golangci/golangci-lint info checking GitHub for tag 'v1.42.0'
golangci/golangci-lint info found version: 1.42.0 for v1.42.0/linux/amd64
golangci/golangci-lint info installed /home/am/go/1.16.0/bin/golangci-lint
>> running golangci-lint
GO111MODULE=on go list -e -compiled -test=true -export=false -deps=true -find=false -tags= -- ./... > /dev/null
go: downloading github.com/go-kit/log v0.2.0
go: downloading github.com/lib/pq v1.10.4
go: downloading github.com/prometheus/client_golang v1.12.1
go: downloading gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: downloading github.com/blang/semver v3.5.1+incompatible
go: downloading github.com/prometheus/common v0.32.1
go: downloading github.com/prometheus/exporter-toolkit v0.7.1
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c
go: downloading github.com/go-logfmt/logfmt v0.5.1
go: downloading github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
go: downloading github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading google.golang.org/protobuf v1.26.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/kr/pretty v0.2.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e
go: downloading github.com/mwitkow/go-conntrack v0.0.0-20190716064945-2f068394615f
go: downloading golang.org/x/net v0.0.0-20210525063256-abc453219eb5
go: downloading golang.org/x/oauth2 v0.0.0-20210514164344-f6687ab2804c
go: downloading golang.org/x/sys v0.0.0-20220114195835-da31bd327af9
go: downloading github.com/kr/text v0.1.0
go: downloading github.com/jpillora/backoff v1.0.0
go: downloading golang.org/x/text v0.3.6
GO111MODULE=on /home/am/go/1.16.0/bin/golangci-lint run  ./...
>> running check for unused/missing packages in go.mod
GO111MODULE=on go mod tidy
go: downloading github.com/stretchr/testify v1.4.0
go: downloading github.com/google/go-cmp v0.5.5
go: downloading google.golang.org/appengine v1.6.6
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
curl -s -L https://github.com/prometheus/promu/releases/download/v0.13.0/promu-0.13.0.linux-amd64.tar.gz | tar -xvzf - -C /tmp/tmp.V7f4P8FhUE
promu-0.13.0.linux-amd64/
promu-0.13.0.linux-amd64/LICENSE
promu-0.13.0.linux-amd64/NOTICE
promu-0.13.0.linux-amd64/promu
mkdir -p /home/am/go/1.16.0/bin
cp /tmp/tmp.V7f4P8FhUE/promu-0.13.0.linux-amd64/promu /home/am/go/1.16.0/bin/promu
rm -r /tmp/tmp.V7f4P8FhUE
>> building binaries
GO111MODULE=on /home/am/go/1.16.0/bin/promu build --prefix /home/am/dev/prj/percona/tasks/PMM-7806-pg-exporter-upgrade/postgres_exporter 
 >   postgres_exporter
>> running all tests
GO111MODULE=on go test -race  ./...
ok      github.com/prometheus-community/postgres_exporter/cmd/postgres_exporter 0.028s
?       github.com/prometheus-community/postgres_exporter/collector     [no test files]
``` 

As you can see all works fine.

Reply:
but I can, change to 1.17
I just don't see a reason for that...

Reply:
it's weird that we have golint in tools/go.mod, but download it using curl. I'm a little bit confused